graspologic/nominate/spectralVN.py

```diff
@@ -0,0 +1,263 @@
+    unique_attributes_ : np.ndarray
+        Each unique attribute represented in the seed. One dimensional. In the unattributed case of SVN, the number of
+        unique attributes (and therefore the shape along axis 0 of `unique_att_`) is equal to the number of seeds ( the
+        shape along axis 0 of `attribute_labels_`).
+    distance_matrix_ : np.ndarray
+        The euclidean distance from each seed vertex to each vertex.
+        Shape is ``(number_vertices, number_unique_attributes)`` if attributed or Shape is
+        ``(number_vertices, number_seed_vertices)`` if unattributed.
+
+    References
+    ----------
+    .. [1] Fishkind, D. E.; Lyzinski, V.; Pao, H.; Chen, L.; Priebe, C. E. Vertex nomination schemes for membership
+        prediction. Ann. Appl. Stat. 9 2015. https://projecteuclid.org/euclid.aoas/1446488749
+
+    .. [2] Jordan Yoder, Li Chen, Henry Pao, Eric Bridgeford, Keith Levin, Donniell E. Fishkind, Carey Priebe,
+        Vince Lyzinski, Vertex nomination: The canonical sampling and the extended spectral nomination schemes,
+        Computational Statistics & Data Analysis, Volume 145, 2020.
+        http://www.sciencedirect.com/science/article/pii/S0167947320300074
+
+    """
+
+    def __init__(
+        self,
+        embedding: np.ndarray = None,
+        embedder: Union[str, BaseSpectralEmbed] = "ASE",
+        persistent: bool = True,
+    ):
+        super().__init__(multigraph=False)
+        self.embedding = embedding
+        if self.embedding is None or not persistent:
+            if isinstance(embedder, BaseSpectralEmbed):
+                self.embedder = embedder
+            elif embedder == "ASE":
+                self.embedder = ase()
+            elif embedder == "LSE":
+                self.embedder = lse()
+            else:
+                raise TypeError
+        elif np.ndim(embedding) != 2:
+            raise IndexError("embedding must have dimension 2")
+        self.persistent = persistent
+        self.attribute_labels_ = None
+        self.unique_attributes_ = None
+        self.distance_matrix_ = None
+        self.neigh_inds = None
+
+    @staticmethod
+    def _make_2d(arr: np.ndarray) -> np.ndarray:
+        # ensures arr is two or less dimensions.
+        # if 1d, adds unique at each index on
+        # the second dimension.
+        if not np.issubdtype(arr.dtype, np.integer):
+            raise TypeError("Argument must be of type int")
+        arr = np.array(arr, dtype=np.int)
+        if np.ndim(arr) > 2 or (arr.ndim == 2 and arr.shape[1] > 2):
+            raise IndexError("Argument must have shape (n) or (n, 1) or (n, 2).")
+        elif np.ndim(arr) == 1 or arr.shape[1] == 1:
+            arr = arr.reshape(-1, 1)
+            arr = np.concatenate((arr, np.arange(arr.shape[0]).reshape(-1, 1)), axis=1)
+        return arr
+
+    def _embed(self, X: np.ndarray):
+        if not self.multigraph:
```

Comment:
looks like this is where you are doing check's on `embedding`. Make sure you also check that it's an np.array, since that's specified